server/src/instant/db/instaql.clj

```diff
@@ -177,32 +187,37 @@
   "Splits keys into segments."
   [state [k v :as c]]
   (cond (or-where-cond? c)
-        {:or (let [conds (->> v
-                              combine-or-where-conds
-                              (map (fn [conds]
-                                     (map (partial coerce-where-cond state)
-                                          (collapse-or-where-conds conds)))))]
-               (if (seq conds)
-                 conds
-                 (ex/throw-validation-err!
-                  :query
-                  (:root state)
-                  [{:expected 'non-empty-list?
-                    :in (conj (:in state) :or)
-                    :message "The list of `or` conditions can't be empty."}])))}
+        (let [conds (->> v
```

Comment:
This is still overly complicated--I think it would be a lot easier to reason about if we got rid of the spec coercion, but that's a bigger project.

Reply:
Oo, that could be interesting! Would make it nicer too when we print stuff deep inside the codebase